.github/workflows/ci.yaml

```
@@ -72,3 +77,55 @@
+        with:
+          python-version: 3.12
+
+      - name: Start Uvicorn Server in Background
+        run: |
+          cd backend
+          pip install poetry
+          poetry install
+          nohup poetry run uvicorn main:app --host 0.0.0.0 --port 8000 &
+
+      - name: Wait for Server to Start
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2181:style:8:6: Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?

(shellcheck)